### PR TITLE
Fix/remove schema evolution

### PR DESCRIPTION
**Pull request by VShkaberda**
### Description

Add source and destination schemes comparison. If column names don't match strictly raise `SchemaNotMatchException`.

### Checklist

- [x] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the README.md file

### CODE REVIEW

File: src/main/scala/mirroring/services/writer/DeltaService.scala [hunk 1]
```diff
@@ -16,13 +16,18 @@
...
+import io.delta.tables.DeltaTable
 import mirroring.builders.FilterBuilder
+import mirroring.services.SparkService.spark
 import org.apache.spark.sql.{DataFrame, DataFrameWriter, Row}
 import wvlet.log.LogSupport
 
 class DeltaService(context: WriterContext) extends LogSupport {
 
   def write(data: DataFrame): Unit = {
+    if (DeltaTable.isDeltaTable(spark, context.path)) {
```
Comment (hanna-liashchuk) [inline src/main/scala/mirroring/services/writer/DeltaService.scala:28]: is this check needed? AFAIK, if mergeSchema is false, [schema enforcement](https://delta.io/blog/2022-11-16-delta-lake-schema-enforcement/) is in place.

File: src/main/scala/mirroring/services/writer/SchemaNotMatchException.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mirroring.services.writer
+
+/** Exception thrown when a schema does not match for source and destination.
+  */
+private[writer] class SchemaNotMatchException() extends Exception
```
Comment (hanna-liashchuk) [inline src/main/scala/mirroring/services/writer/SchemaNotMatchException.scala:21]: maybe let's move this class to an `exceptions` package?

File: src/main/scala/mirroring/services/writer/ChangeTrackingService.scala [hunk 3]
```diff
@@ -33,11 +33,14 @@ class ChangeTrackingService(
...
     s"${Config.SourceAlias}.SYS_CHANGE_OPERATION in ('I', 'U')"
+  private val sourceColPrefix = "SYS_CHANGE_PK_"
+  private val excludeColumns =
+    context.primaryKey.map(col => s"$sourceColPrefix$col") :+ "SYS_CHANGE_OPERATION"
 
   override def write(data: DataFrame): Unit = {
     if (DeltaTable.isDeltaTable(spark, context.path)) {
-
       logger.info("Target table already exists. Merging data...")
+      verifySchemaMatch(data)
```
Comment (hanna-liashchuk) [inline src/main/scala/mirroring/services/writer/ChangeTrackingService.scala:43]: according to the [docs](https://docs.databricks.com/delta/update-schema.html#automatic-schema-evolution-for-delta-lake-merge) and tests that I performed, merge is not merging schema by default. it fails with an error if a column is altered or ignore a new column

**Review by hanna-liashchuk — CHANGES_REQUESTED** [review]

File: src/main/scala/mirroring/services/writer/DeltaService.scala (hunk 1 above)
Comment (VShkaberda) [reply src/main/scala/mirroring/services/writer/DeltaService.scala:28]: Yes, it is needed. If the table is mirrored at the first time, one doesn't have a delta table, so `AnalysisException: Path does not exist` is raised. Another way is `try .. catch`  block catching `case e: org.apache.spark.sql.AnalysisException if e.getMessage().startsWith("...` but it seems to be less preferred.

File: src/main/scala/mirroring/services/writer/ChangeTrackingService.scala (hunk 3 above)
Comment (VShkaberda) [reply src/main/scala/mirroring/services/writer/ChangeTrackingService.scala:43]: According to the docs:

> Any columns in the source dataset that don’t match columns in the target table are ignored.

Our demand is to block merge if there is columns in the source dataset that don’t match columns in the target.

Comment (hanna-liashchuk) [reply src/main/scala/mirroring/services/writer/ChangeTrackingService.scala:43]: indeed, ok

File: src/main/scala/mirroring/services/writer/SchemaNotMatchException.scala (hunk 2 above)
Comment (VShkaberda) [reply src/main/scala/mirroring/services/writer/SchemaNotMatchException.scala:21]: I got this idea from [spark](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/TaskNotSerializableException.scala) library. This approach has two benefits:
  - exception remains private to `writer`
  - exception needs no import statement inside `mirroring.services.writer`

**Review by hanna-liashchuk — APPROVED** [review]

**Pull request merged**